src/mixins/collectionMixin.js

```diff
@@ -466,9 +467,12 @@ const apollo = {
     // Optional result hook
     result({ data, loading, networkStatus }) {
       if (data && data.datasets) {
-        this.currentDataset = data.datasets[0];
-        this.fieldsCount = this.currentDataset.fields.length;
-        this.combinatorsCount = this.currentDataset.combinators.length;
+        [this.currentDataset] = data.datasets;
+        if (this.$route.name === 'Update Combinator') {
+          this.datasets = [this.currentDataset]
+        }
+        this.fieldsCount = this.currentDataset.fields_count;
```

Comment:
These are changed based on the `currentDataset` even if `this.datasets` isn't changed. Is that correct? (not saying it's not - just curious)

Reply:
Correct. I only set datasets to contain currentDataset, when updating a combinator, so that it shows in the vue form selection box. 
The alternative is a query to get ALL datasets, in order to just show the name of the one being used.